File: sdks/ruby/lib/novabilling/charges/types/update_charge_dto.rb
# frozen_string_literal: true

module Novabilling
  module Charges
    module Types
      class UpdateChargeDto < Internal::Types::Model
        field :id, -> { String }, optional: false, nullable: false
        field :billing_timing, -> { Novabilling::Charges::Types::UpdateChargeDtoBillingTiming }, optional: true, nullable: false, api_name: "billingTiming"
        field :invoice_display_name, -> { String }, optional: true, nullable: false, api_name: "invoiceDisplayName"
        field :min_amount_cents, -> { Integer }, optional: true, nullable: false, api_name: "minAmountCents"
        field :prorated, -> { Internal::Types::Boolean }, optional: true, nullable: false
        field :properties, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false
        field :graduated_ranges, -> { Internal::Types::Array[Novabilling::Types::GraduatedRangeDto] }, optional: true, nullable: false, api_name: "graduatedRanges"
        field :filters, -> { Internal::Types::Array[Novabilling::Types::ChargeFilterDto] }, optional: true, nullable: false
      end
    end
  end
end

File: sdks/php/src/Charges/Requests/UpdateChargeDto.php
<?php

namespace NovaBilling\Charges\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Charges\Types\UpdateChargeDtoBillingTiming;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Core\Types\ArrayType;
use NovaBilling\Types\GraduatedRangeDto;
use NovaBilling\Types\ChargeFilterDto;

class UpdateChargeDto extends JsonSerializableType
{
    /**
     * @var ?value-of<UpdateChargeDtoBillingTiming> $billingTiming
     */
    #[JsonProperty('billingTiming')]
    public ?string $billingTiming;

    /**
     * @var ?string $invoiceDisplayName
     */
    #[JsonProperty('invoiceDisplayName')]
    public ?string $invoiceDisplayName;

    /**
     * @var ?float $minAmountCents
     */
    #[JsonProperty('minAmountCents')]
    public ?float $minAmountCents;

    /**
     * @var ?bool $prorated
     */
    #[JsonProperty('prorated')]
    public ?bool $prorated;

    /**
     * @var ?array<string, mixed> $properties
     */
    #[JsonProperty('properties'), ArrayType(['string' => 'mixed'])]
    public ?array $properties;

    /**
     * @var ?array<GraduatedRangeDto> $graduatedRanges
     */
    #[JsonProperty('graduatedRanges'), ArrayType([GraduatedRangeDto::class])]
    public ?array $graduatedRanges;

    /**
     * @var ?array<ChargeFilterDto> $filters
     */
    #[JsonProperty('filters'), ArrayType([ChargeFilterDto::class])]
    public ?array $filters;

    /**
     * @param array{
     *   billingTiming?: ?value-of<UpdateChargeDtoBillingTiming>,
     *   invoiceDisplayName?: ?string,
     *   minAmountCents?: ?float,
     *   prorated?: ?bool,
     *   properties?: ?array<string, mixed>,
     *   graduatedRanges?: ?array<GraduatedRangeDto>,
     *   filters?: ?array<ChargeFilterDto>,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->billingTiming = $values['billingTiming'] ?? null;
        $this->invoiceDisplayName = $values['invoiceDisplayName'] ?? null;
        $this->minAmountCents = $values['minAmountCents'] ?? null;
        $this->prorated = $values['prorated'] ?? null;
        $this->properties = $values['properties'] ?? null;
        $this->graduatedRanges = $values['graduatedRanges'] ?? null;
        $this->filters = $values['filters'] ?? null;
    }
}
